File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.go
func findKthPositive(arr []int, k int) int {
    low := 0
    high := len(arr)-1
    var  mid,missing int
    for low<=high{
        mid = (low+high)/2 
        missing = arr[mid]-(mid+1)
        if(missing < k){
            low = mid+1
        }else{
            high = mid-1
        }
    }
    return low+k
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int low = 0, high = arr.length-1, mid, missing;
        while(low<=high){
            mid = (low+high)/2;
            missing = arr[mid]-(mid+1);
            if(missing < k)
                low = mid+1;
            else
                high = mid-1;
        }
        return low+k;
    }
}